src/components/SpruceForm/CustomFields.tsx

```diff
@@ -1,22 +1,35 @@
 import styled from "@emotion/styled";
-import { Description, Subtitle } from "@leafygreen-ui/typography";
-import { FieldProps } from "@rjsf/core";
+import { Description, H3, Subtitle } from "@leafygreen-ui/typography";
+import { Field } from "@rjsf/core";
 
-export const TitleField: React.FC<FieldProps> = ({ id, title }) => (
-  <>
-    {/* @ts-expect-error  */}
-    <StyledSubtitle id={id}>{title}</StyledSubtitle>
-  </>
-);
+export const TitleField: Field = ({ id, isSectionTitle, title }) => {
+  const Component = isSectionTitle ? StyledH3 : StyledSubtitle;
+  return (
+    <>
+      {/* @ts-expect-error  */}
+      <Component id={id}>{title}</Component>
+    </>
```

Comment:
nit: it might be more readable if we named this `Component` variable something like `TitleFieldTag` or `TitleTag`